wire-ios/Wire-iOS/Sources/UserInterface/Location/AppLocationManager.swift

```diff
@@ -0,0 +1,80 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CoreLocation
+
+// MARK: - AppLocationManagerDelegate
+
+protocol AppLocationManagerDelegate: AnyObject {
+    func didUpdateLocations(_ locations: [CLLocation])
+    func didFailWithError(_ error: Error)
+    func didChangeAuthorization(status: CLAuthorizationStatus)
+}
+
+class AppLocationManager: NSObject {
+
+    // MARK: - Properties
+
+    static let shared = AppLocationManager()
```

Comment:
How hard would it be to avoid using a singleton?

Reply:
@johnxnguyen Done! https://github.com/wireapp/wire-ios/commit/1a8aaf2b62d71a5a770c1bb55833c1ea8f570bd4